src/trace/ord/val_batch.rs

```diff
@@ -393,12 +393,17 @@ where
             self.cursor.child.child.step(&storage.layer.vals.vals);
         }
     }
-    fn weight<'a>(&self, storage: &'a Self::Storage) -> &'a R
+    fn weight(&mut self, storage: &Self::Storage) -> R
```

Comment:
Looks like we clone within pretty much every `.weight()` call, maybe it should return a reference and allow the caller to decide whether or not it should be cloned

Reply:
Problem is, the result sometimes needs to be computed, but I can probably make it work...

Reply:
Actually, no, not in a nice way.